skglm/tests/test_prox_newton.py

```diff
@@ -20,13 +20,13 @@ def test_alpha_max(X_density):
 
```

Comment:
alpha_max changes when fitting intercept

Reply:
It is ``alpha_max`` for the penalized coefficients. Hence adding an unpenalized intercept doesn't change it.

If I understand what you mean: the condition to have a ``0`` intercept translates to having a centered ``y``.

Reply:
That would be the case for the Lasso, but for Logreg things are different. 
My point is, you compute the alpha_max without intercept, then fit with an intercept, so without more computations, it's not clear what should happen (should we expect 0 as a solution or not?)

Reply:
I 100% agree with you! Yet, in terms of ``alpha_max``, Lasso resemble slightly LogReg as $\nabla F(0) = \frac{y}{2 n}$

$$ -\mathbf{X}^\top \nabla F(0) \in \lambda \partial \lVert \cdot \rVert_1 \quad \mathbb{1}_n^\top \nabla F(0) = 0$$

$$   \lVert \frac{\mathbf{X}^\top y}{2 n}  \rVert_\infty \leq \lambda, \quad \bar{y} = 0$$

I have added a comment in the unittest to clarify that.

Reply:
I have re-checked. There are some subtleties in the intercept case.
I resolved to exclude ``fit_intercept=True`` from the ``alpha_max`` unit test.

What do you think @mathurinm?

Reply:
I don't think it's a pathological case as ``alpha_max`` changes when fitting an unpenalized intercept.

I can show that if we set ``alpha_max = norm(X.T @ y, ord=np.inf) / n_samples`` (remove 2 from denominator),
we would have ``w[:n_features] == 0`` in both cases (``fit_intercept`` ``True`` or ``False``), 

However, I don't think it's a clever idea to refactor this unittest to handle ``fit_intercept=True`` and write a long math justification to explain it. Indeed, we have a second unittest that checks thoroughly the case ``fit_intercept=True``

Reply:
I agree, it is not worth the complication. Tests should make our life easier not complicate the codebase :)